### PR TITLE
Prometheus: Migrate to use SDK contracts

### PR DESCRIPTION
**Pull request by dsotirakis**
**What this PR does / why we need it**:

This PR is responsible for the migration of Prometheus to start using the backend plugin SDK

**Which issue(s) this PR fixes**:

Fixes (a checkbox in) #35179

### CODE REVIEW

**Review by ivanahuckova — COMMENTED** [review]
🚀🚀🚀🚀Awesome job @dsotirakis! I've left a question regarding **customQueryParametersMiddleware** below. 🙂

File: pkg/tsdb/prometheus/prometheus.go [hunk 1]
```diff
@@ -2,73 +2,136 @@ package prometheus
...
+	if err := s.BackendPluginManager.Register("prometheus", factory); err != nil {
+		plog.Error("Failed to register plugin", "error", err)
+	}
+	return nil
 }
 
-//nolint: staticcheck // plugins.DataPlugin deprecated
-func New(provider httpclient.Provider) func(*models.DataSource) (plugins.DataPlugin, error) {
-	return func(dsInfo *models.DataSource) (plugins.DataPlugin, error) {
-		transport, err := dsInfo.GetHTTPTransport(provider, customQueryParametersMiddleware(plog))
```
[the remark this answers is not in the document]
Comment (dsotirakis) [reply pkg/tsdb/prometheus/prometheus.go:40]: Ooooh missed that completely - pushed another commit, could you test it again please? 🙇

**Review by ivanahuckova — APPROVED** [review]
LGTM 🚀. Great job @dsotirakis! 🤩 
 
One minor question, do we still need those `//nolint: staticcheck // plugins.DataResponse deprecated` comments. I think that they could be removed now.

**Comment by ivanahuckova** [comment]
@dsotirakis I have merged https://github.com/grafana/grafana/pull/36796 which created merge conflicts with your PR. Let me know if I can help with resolving them. 🙂

**Pull request merged**